internal/observability/middleware_test.go

```diff
@@ -85,8 +91,11 @@ func TestMiddlewareLogging(t *testing.T) {
 			},
 		},
 		{
-			desc: "downstream transport error",
-			err:  failed,
+			desc:            "downstream transport error",
```

Comment:
We should add a test case here for application errors. In that case, `err` is nil but `applicationErr` is true.